orc8r/cloud/go/syncstore/store_reader.go

```diff
@@ -0,0 +1,264 @@
+/*
+ Copyright 2020 The Magma Authors.
+
+ This source code is licensed under the BSD-style license found in the
+ LICENSE file in the root directory of this source tree.
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package syncstore
+
+import (
+	"database/sql"
+	"encoding/binary"
+	"fmt"
+
+	"magma/orc8r/cloud/go/blobstore"
+	configurator_storage "magma/orc8r/cloud/go/services/configurator/storage"
+	"magma/orc8r/cloud/go/sqorc"
+	"magma/orc8r/cloud/go/storage"
+	merrors "magma/orc8r/lib/go/errors"
+	"magma/orc8r/lib/go/protos"
+
+	"github.com/Masterminds/squirrel"
+	"github.com/golang/protobuf/proto"
+	"github.com/pkg/errors"
+)
+
+const (
+	// idCol contains the network-wide unique identifiers of the objects.
+	idCol              = "id"
+	nidCol             = "network_id"
+	rootDigestCol      = "root_digest"
+	leafDigestsCol     = "leaf_digests"
+	objCol             = "obj"
+	lastUpdatedTimeCol = "last_updated_at"
+
+	lastResyncBlobstoreType  = "gateway_last_resync_time"
+	cacheWriterBlobstoreType = "cache_writer_creation_time"
+)
+
+func NewSyncStoreReader(db *sql.DB, builder sqorc.StatementBuilder, fact blobstore.BlobStorageFactory, config Config) (SyncStoreReader, error) {
+	err := config.Validate(false)
+	if err != nil {
+		return nil, errors.Wrap(err, "invalid configs for syncstore reader")
+	}
+	storeReader := &syncStore{
+		db: db, builder: builder, fact: fact,
```

Comment:
Here and elsewhere: eh just make these separate lines, usually I don't like mixing single vs multiple per-line